mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageId.java

```diff
@@ -26,8 +26,12 @@
         MessageId fromString(String serialized);
 
         MessageId generate();
+
+        boolean canDeserialize();
```

Comment:
One other way might be to check MailboxManager capability.

But I don't like the id of MailboxManager depending on SearchIndex depending on MailboxManager (more : I fear for Spring)

We are inside the mailbox. Does capabilities as a all have sens ?